### PR TITLE
add benchmark memory profiles into report

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

**What this PR does / why we need it**:

giving a detailed report of memory usage, preview:

image

**Which issue(s) this PR fixes**:

xref: https://github.com/envoyproxy/gateway/issues/3698

### CODE REVIEW

**Comment by Xunzhuo** [comment]
@shawnh2 plz fix the lint errors

**Comment by shawnh2** [comment]
> @shawnh2 plz fix the lint errors

Done! PTAL!

File: tools/make/kube.mk [hunk 1]
```diff
@@ -79,6 +79,7 @@ kube-deploy-for-benchmark-test: manifests helm-generate ## Install Envoy Gateway
 	helm install eg charts/gateway-helm --set deployment.envoyGateway.imagePullPolicy=$(IMAGE_PULL_POLICY) \
 		--set deployment.envoyGateway.resources.limits.cpu=$(BENCHMARK_CPU_LIMITS) \
 		--set deployment.envoyGateway.resources.limits.memory=$(BENCHMARK_MEMORY_LIMITS) \
+		--set config.envoyGateway.admin.enablePprof=true \
```
Comment (arkodg) [inline tools/make/kube.mk:82]: curious if setting this know affected existing benchmark throughput ?

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zhaohuabing — DISMISSED** [review]
🚀

**Comment by shawnh2** [comment]
Hold for now, need to investigate why the benchmark test in CI is failing.

File: tools/make/kube.mk (hunk 1 above)
Comment (shawnh2) [reply tools/make/kube.mk:82]: compare two benchmark results: https://github.com/envoyproxy/gateway/actions/runs/10163594751/job/28107439647?pr=3951 (added) & https://github.com/envoyproxy/gateway/actions/runs/10097739129/job/27970475986?pr=3682 (not added), the results almost the same.

**Review by cnvergence — APPROVED** [review]

**Review by arkodg — APPROVED** [review]

**Pull request merged**